src/js/mixins/evented.js

```diff
@@ -0,0 +1,385 @@
+  (Array.isArray(type) && !!type.length);
+
+/**
+ * Validates a value to determine if it is a valid event target. Throws if not.
+ *
+ * @throws {Error}
+ *         If the target does not appear to be a valid event target.
+ *
+ * @param  {Object} target
+ *         The object to test.
+ */
+const validateTarget = (target) => {
+  if (!target.nodeName && !isEvented(target)) {
+    throw new Error('Invalid target; must be a DOM node or evented object.');
+  }
+};
+
+/**
+ * Validates a value to determine if it is a valid event target. Throws if not.
+ *
+ * @throws {Error}
+ *         If the type does not appear to be a valid event type.
+ *
+ * @param  {string|Array} type
+ *         The type to test.
+ */
+const validateEventType = (type) => {
+  if (!isValidEventType(type)) {
+    throw new Error('Invalid event type; must be a non-empty string or array.');
+  }
+};
+
+/**
+ * Validates a value to determine if it is a valid listener. Throws if not.
+ *
+ * @throws {Error}
+ *         If the listener is not a function.
+ *
+ * @param  {Function} listener
+ *         The listener to test.
+ */
+const validateListener = (listener) => {
+  if (typeof listener !== 'function') {
+    throw new Error('Invalid listener; must be a function.');
+  }
+};
+
+/**
+ * Takes an array of arguments given to `on()` or `one()`, validates them, and
+ * normalizes them into an object.
+ *
+ * @private
+ * @param  {Object} self
+ *         The evented object on which `on()` or `one()` was called.
+ *
+ * @param  {Array} args
+ *         An array of arguments passed to `on()` or `one()`.
+ *
+ * @return {Object}
+ *         An object containing useful values for `on()` or `one()` calls.
+ */
+const normalizeListenArgs = (self, args) => {
+
+  // If the number of arguments is less than 3, the target is always the
+  // evented object itself.
+  const isTargetingSelf = args.length < 3 || args[0] === self || args[0] === self.eventBusEl_;
+  let target;
+  let type;
+  let listener;
+
+  if (isTargetingSelf) {
+    target = self.eventBusEl_;
+
+    // Deal with cases where we got 3 arguments, but we are still listening to
+    // the evented object itself.
+    if (args.length >= 3) {
+      args.shift();
+    }
+
+    [type, listener] = args;
+  } else {
+    [target, type, listener] = args;
+  }
+
+  validateTarget(target);
+  validateEventType(type);
+  validateListener(listener);
+
+  listener = Fn.bind(self, listener);
+
+  return {isTargetingSelf, target, type, listener};
+};
+
+/**
+ * Adds the listener to the event type(s) on the target, normalizing for
+ * the type of target.
+ *
+ * @private
+ * @param  {Element|Object} target
+ *         A DOM node or evented object.
+ *
+ * @param  {string|Array} type
+ *         One or more event type(s).
+ *
+ * @param  {Function} listener
+ *         A listener function.
+ *
+ * @param  {string} [method="on"]
+ *         The event binding method to use.
+ */
+const listen = (target, type, listener, method = 'on') => {
```

Comment:
Sounds good. Also moving `method` to be the second argument to better reflect the usage.